src/qml/bitcoin.cpp

```diff
@@ -158,6 +160,188 @@ void setupChainQSettings(QGuiApplication* app, QString chain)
         app->setApplicationName(QAPP_APP_NAME_REGTEST);
     }
 }
+
+// added this function to set custom data directory when starting node
+bool setCustomDataDir(QString strDataDir)
+{
+    if(fs::exists(GUIUtil::QStringToPath(strDataDir))){
+        gArgs.SoftSetArg("-datadir", fs::PathToString(GUIUtil::QStringToPath(strDataDir)));
+        gArgs.ClearPathCache();
+    return true;
+    } else {
+        return false;
+    }
+}
+
+QGuiApplication* m_app;
```

Comment:
statically defining all of these pointers is concerning. Eventhough it is redundant it feels safer to keep a lot of these withing the function scopes.

Reply:
Ok thanks for looking into it... will refactor further to use arguments instead of pointers

Reply:
After our online chat, let's keep things as is with the caveat that if any issues arise due to the static pointers we'll have to refactor further. Resolving for now...

Reply:
> After our online chat...

Could the main points from that discussion be posted here?

Reply:
To summarize our convo: We talked about the safety of static pointers in the code. We thought about adding a class like in #390, however decided to leave them for now, since there might be [issues](https://github.com/bitcoin-core/gui-qml/pull/390#discussion_r1531658261) raised by @Jarold that we need to consider first. That said we both agreed that having a clear owner for pointers is usually a good idea for better code management. Is there anything I missed @johnny9 ?